frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.component.ts

```diff
@@ -147,6 +164,41 @@ export class ModelDiagramDialogComponent {
         saveAs(response, `${uuid}.svg`);
       });
   }
+
+  async getUser() {
+    const token = localStorage.getItem('access_token');
+
+    const headers = this.getHeaders(token);
+
+    const response = await fetch(environment.backend_url + '/users/current', {
+      headers,
+    });
+    return response.json();
+  }
+
+  async insertToken() {
+    const user = await this.getUser();
+    const id = user.id;
+    const token = localStorage.getItem('access_token');
+    this.isButtonClicked = true;
+    const headers = this.getHeaders(token);
+
+    const response = await fetch(
+      environment.backend_url + `/users/${id}/token`,
+      {
+        headers,
+      }
+    );
+    const basic_token = await response.text();
+    this.passwordValue = basic_token.replaceAll('"', '');
+  }
```

Comment:
Here I would move the actual http call to the `UserService` which is responsible for any user related actions (e.g., creating a user, getting the current user, deleting a user). There, we can add a `createToken` function (or any other fitting name) that could look like this:
```typescript
createToken(): Observable<string> {
    return this.http.post<string>(
      this.BACKEND_URL_PREFIX + `current/token`,
      {}
    );
  }
```

In here, we then just have to put in the following:
```typescript
insertToken() {
  this.userService
      .createToken()
      .subscribe((token) => (this.passwordValue = token.replaceAll('"', '')));
}
```